idseq_pipeline/commands/accessionid2seq_functions.py

```diff
@@ -95,48 +95,68 @@ def generate_alignment_viz_json(nt_file, nt_loc_db, db_type,
 
     result_dict = {}
     to_be_deleted = []
+    error_count = 0
     for accession_id, ad in groups.iteritems():
-        tmp_file = 'accession-%s' % accession_id
-        if ad['ref_seq_len'] <= MAX_SEQ_DISPLAY_SIZE and 'ref_seq' not in ad:
-            with open(tmp_file, "rb") as tf:
-                ad['ref_seq'] = tf.read()
-                # the previous value of ad['ref_seq_len'] is greater because it includes newline characters
-                # the value we set here excludes newline characters
-                ad['ref_seq_len'] = len(ad['ref_seq'])
-            to_be_deleted.append(tmp_file)
-        if 'ref_seq' in ad:
-            ref_seq = ad['ref_seq']
-            for read in ad['reads']:
-                prev_start, ref_start, ref_end, post_end = read[3]
-                read[3] = [ref_seq[prev_start:ref_start], ref_seq[ref_start:ref_end], ref_seq[ref_end:post_end]]
-        else:
-            # the reference sequence is too long to read entirely in RAM, so we only read the mapped segments
+        try:
             tmp_file = 'accession-%s' % accession_id
-            with open(tmp_file, "rb") as tf:
+            if ad['ref_seq_len'] <= MAX_SEQ_DISPLAY_SIZE and 'ref_seq' not in ad:
+                if ad['ref_seq_len'] == 0:
+                    ad['ref_seq'] = "REFERENCE SEQUENCE NOT FOUND"
```

Comment:
This is the only added code, aside from exception handling and error counting.    Also, this was the only real problem.   I guess the exception handling and error handling aren't really needed, but since I coded and tested, why not.